src/lsm/groove.zig

```diff
@@ -387,12 +389,16 @@ pub fn GrooveType(
 
         const primary_field = if (has_id) "id" else "timestamp";
         const PrimaryKey = @TypeOf(@field(@as(Object, undefined), primary_field));
-        const PrefetchKey = struct { key: union(enum) {
-            id: PrimaryKey,
-            timestamp: u64,
-        }, level: u8 };
+        const PrefetchKey = struct {
+            key: union(enum) {
+                id: PrimaryKey,
+                timestamp: u64,
+            },
+            fingerprint: Fingerprint,
+            level: u8,
+        };
 
-        const PrefetchKeys = std.AutoHashMapUnmanaged(PrefetchKey, void);
+        const PrefetchKeys = std.AutoArrayHashMapUnmanaged(PrefetchKey, void);
```

Comment:
I've tried that, not changed, just measurement noise.
Maybe it's a good design for correctness since it should be unique per `union(enum) {id,timestamp}` and not by the whole struct.